CHANGES.md

```diff
@@ -10,6 +10,11 @@
     an estimator through the `estimator` parameter. To use a pre-fitted
     estimator, pass `prefit=True`.
 * Rename arguments of `create_group_metric_set()` to match the dashboard
+* Remove `Reduction` base class for reductions methods and replace it with
+  `sklearn.base.BaseEstimator`, `sklearn.base.ClassifierMixin`, and
```

Comment:
What functionality do these add, and should there be a smoke test to verify that such functionality really is present?

Reply:
You can eventually pass the sklearn tests as whone [here](https://scikit-learn.org/stable/auto_examples/release_highlights/plot_release_highlights_0_22_0.html#checking-scikit-learn-compatibility-of-an-estimator).

``` python
from sklearn.linear_model import LogisticRegression
from sklearn.tree import DecisionTreeRegressor
from sklearn.utils.estimator_checks import parametrize_with_checks


@parametrize_with_checks([LogisticRegression, DecisionTreeRegressor])
def test_sklearn_compatible_estimator(estimator, check):
    check(estimator)
```

Reply:
That would require a few more changes to work, right?

Reply:
Yes, I'm working on adding the test for ThresholdOptimizer, and once that's done, I guess we'd all have a better idea of what needs to be changed for it to work. In the mean time, this may help understanding what needs to be done:

https://fosdem.org/2020/schedule/event/python2020_scikit_learn_estimator/